examples/fed-pubsub.cc

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright (c) 2017-2019, Battelle Memorial Institute; Lawrence Livermore National Security, LLC; Alliance for Sustainable Energy, LLC.  See the top-level NOTICE for additional details. All rights reserved.
+
+SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#include "helics/application_api/ValueFederate.hpp"
+#include "helics/application_api/Inputs.hpp"
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <regex>
+#include <set>
+#include <thread>
+#include <stdexcept>
+#include "helics/core/helicsCLI11.hpp"
+#include "helics/core/helicsVersion.hpp"
+
+int main (int argc, char *argv[])
+{
+    helics::helicsCLI11App app ("PubSub NS3 Example", "PubSubNS3Example");
+    std::string myname = "fed";
+    std::string publishKey = "ns3_test_value";
+
+    app.add_option ("--name,-n", myname, "name of this federate");
+    app.add_option ("--key,-k", publishKey, "name of the key to publish on");
+
+    auto ret = app.helics_parse (argc, argv);
+
+    helics::FederateInfo fi {};
+    if (ret == helics::helicsCLI11App::parse_output::help_call)
+    {
+        fi.loadInfoFromArgs ("--help");
+        return 0;
+    }
+    else if (ret != helics::helicsCLI11App::parse_output::ok)
+    {
+        return -1;
+    }
+
+    fi.loadInfoFromArgs (argc, argv);
+    fi.setProperty (helics_property_int_log_level, 5);
+
+    auto mFed = std::make_unique<helics::ValueFederate> (myname, fi);
```

Comment:
Attempting to use `HelicsHelper` instead of directly calling `fi.loadInfoFromArgs` results in the following error:
```
3371-aoVGb-J78nH-DZKJF-5F4uI (0)::broker responded with error for registration of ns3::duplicate federate name detected
terminate called after throwing an instance of '
helics::RegistrationFailure'
  what():  fed received Failure duplicate federate name detected
```
I don't know enough about HELICS to really know what's happening, but it seems like this is somehow causing it to register the same federate twice?  For reference, here's the relevant bit of code:
```cpp
int main (int argc, char *argv[])
{
    using ns3::helics_federate;
    helics::helicsCLI11App app ("PubSub NS3 Example", "PubSubNS3Example");
    std::string publishKey = "ns3_test_value";

    app.add_option ("--key,-k", publishKey, "name of the key to publish on");

    const auto ret = app.helics_parse (argc, argv);

    if (ret == helics::helicsCLI11App::parse_output::help_call)
    {
        helics::FederateInfo fi {};
        fi.loadInfoFromArgs ("--help");
        return 0;
    }
    else if (ret != helics::helicsCLI11App::parse_output::ok)
    {
        return -1;
    }

    ns3::HelicsHelper helicsHelper;
    helicsHelper.SetupFederate (argc, argv); // Error results from here
    // ... the rest of main ...
}
```

Reply:
Yea, the expectation is that typically (at least for the examples) there would just be one `ns3` federate. The model for the other examples is that the `ns3-*` example is the one named `ns3`, and then the `fed-*` has a default name of `fed`, so in some ways with this change `ns3-pubsub` would be a more accurate name then `fed-pubsub`.

In the `ns3-sndrcv` example there is a call to `helicsHelper.SetupCommandLine` that adds a `name` argument that can be used to start `ns3-sndrcv` with a different name.